src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceApiServiceSpec.scala

```diff
@@ -27,159 +26,57 @@ class WorkspaceApiServiceSpec extends BaseServiceSpec with WorkspaceApiService w
-          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
-      )
-    // workspaces/%s/%s/unlock
-    workspaceServer
-      .when(request().withMethod("PUT").withPath(unlockPath))
-      .respond(
-        org.mockserver.model.HttpResponse.response()
-          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
-      )
-    // workspaces/%s/%s/checkBucketReadAccess
-    workspaceServer
-      .when(request().withMethod("GET").withPath(bucketPath))
-      .respond(
-        org.mockserver.model.HttpResponse.response()
-          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
-      )
-
-    // Entity responders
-    workspaceServer
-      .when(
-        request()
-          .withMethod("PATCH")
-          .withPath(s"${workspaceBasePath}/${workspace.namespace.get}/${workspace.name.get}")
-          .withHeader(authHeader))
-      .respond(
-        org.mockserver.model.HttpResponse.response()
-          .withHeaders(MockUtils.header)
-          .withStatusCode(NoContent.intValue)
-          .withBody(rawlsErrorReport(NoContent).toJson.compactPrint)
-      )
+  var rawlsServer: ClientAndServer = _
+
+  /** Stub the mock rawls service to respond to a request. Used for testing passthroughs.
+    *
+    * @param method HTTP method to respond to
+    * @param path   request path
+    * @param status status for the response
+    */
+  def stubRawlsService(method: HttpMethod, path: String, status: StatusCode): Unit = {
+    rawlsServer
+      .when(request().withMethod(method.name).withPath(path))
+      .respond(org.mockserver.model.HttpResponse.response()
+        .withHeaders(MockUtils.header).withStatusCode(status.intValue))
+  }
 
-    workspaceServer
-      .when(
-        request()
-          .withMethod("POST")
-          .withPath(s"${workspaceBasePath}/${workspace.namespace.get}/${workspace.name.get}/entities/batchUpsert")
-          .withHeader(authHeader))
+  def stubRawlsServiceWithError(method: HttpMethod, path: String, status: StatusCode) = {
+    rawlsServer
+      .when(request().withMethod(method.name).withPath(path).withHeader(authHeader))
       .respond(
         org.mockserver.model.HttpResponse.response()
           .withHeaders(MockUtils.header)
-          .withStatusCode(NoContent.intValue)
-          .withBody(rawlsErrorReport(NoContent).toJson.compactPrint)
+          .withStatusCode(status.intValue)
+          .withBody(rawlsErrorReport(status).toJson.compactPrint)
       )
+  }
 
-    workspaceServer
-      .when(
-        request()
-          .withMethod("POST")
-          .withPath(s"${workspaceBasePath}/${workspace.namespace.get}/${workspace.name.get}/entities/batchUpdate")
-          .withHeader(authHeader))
-      .respond(
-        org.mockserver.model.HttpResponse.response()
-          .withHeaders(MockUtils.header)
-          .withStatusCode(NoContent.intValue)
-          .withBody(rawlsErrorReport(NoContent).toJson.compactPrint)
-      )
+  override def beforeEach(): Unit = {
```

Comment:
Exactly. The errors were very spurious - the only way to reproduce was to restart some variable number of times.